### ISSUE TITLE
SurrealDB incompatibility due to WebSocket client

### ISSUE THREAD

**Issue by Autumnlight02** [oven-sh/bun#5434]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Linux 6.3.12-100.fc37.x86_64 x86_64 x86_64

### What steps can reproduce the bug?

https://github.com/surrealdb/surrealdb.js/issues/159

Basically the goal was to test up a surrealDB database connection, which breaks shortly after, the issue above mentiones it already.

Here is a small guide on how to reproduce it.
1. [assuming you have the latest surrealDB] surreal start --log trace --user root --pass root file:mydatabase.db
Run:
`import Surreal from "surrealdb.js";

const db = new Surreal();
console.log("b");

async function main() {
  try {
    // Connect to the database
    await db.connect("http://127.0.0.1:8000/rpc");

    // Signin as a namespace, database, or root user
    await db.signin({
      user: "root",
      pass: "root",
    });
    console.log("works");

    // Select a specific namespace / database
    await db.use({ ns: "test", db: "test" });
    console.log("fails here");
    } catch(err) {console.log(err)}
    }
    main()
    `
    The database log reads the following:
    `[2023-09-15 05:04:11] INFO  surrealdb::env Running 1.0.0-beta.8+20230320.735856e for linux on x86_64
[2023-09-15 05:04:11] INFO  surrealdb::iam Root authentication is enabled
[2023-09-15 05:04:11] INFO  surrealdb::iam Root username is 'root'
[2023-09-15 05:04:11] INFO  surrealdb::dbs Database strict mode is disabled
[2023-09-15 05:04:11] INFO  surrealdb::kvs Starting kvs store at file:mydatabase.db
[2023-09-15 05:04:11] INFO  surrealdb::kvs Started kvs store at file:mydatabase.db
[2023-09-15 05:04:11] INFO  surrealdb::net Starting web server on 0.0.0.0:8000
[2023-09-15 05:04:11] INFO  surrealdb::net Started web server on 0.0.0.0:8000
[2023-09-15 05:04:14] INFO  surrealdb::net 127.0.0.1:60886 GET /rpc HTTP/1.1 101 "-" 60.729µs
[2023-09-15 05:04:14] TRACE surrealdb::net WebSocket '69acc628-45da-4145-b583-ae5a5d3ae4ad' connected
[2023-09-15 05:04:14] TRACE surrealdb::net RPC Received: { id: '1', method: 'signin', params: [{ pass: 'root', user: 'root' }] }
[2023-09-15 05:04:14] TRACE surrealdb::net WebSocket error: Protocol(UnexpectedContinueFrame)
[2023-09-15 05:04:14] TRACE surrealdb::net WebSocket '69acc628-45da-4145-b583-ae5a5d3ae4ad' disconnected
`

### What is the expected behavior?

That you can simply connect to the db and work with it

### What do you see instead?

That somehow the websocket breaks and that the try catch fails to activate / trigger

### Additional information

Described here: https://github.com/surrealdb/surrealdb.js/issues/159

**Comment by Micnubinub** [comment]
Same issue on `Ubuntu 23.04 Linux 6.2.0-32-generic` and `MacOS 13.5.2` with `Surreal 1.0.0+20230913.54aedcd`

**Comment by happysalada** [comment]
I wonder if this was fixed after https://github.com/oven-sh/bun/pull/6564

**Comment by lkwr** [comment]
@happysalada `Bun v1.0.7` still does not work for me with `surrealdb.js@0.9.1` :(

**Comment by Electroid** [comment]
We need to test our WebSocket client using the [Autobahn](https://github.com/crossbario/autobahn-testsuite) test suite to fix these bugs, including this one. For updates, please upvote and subscribe to this issue: https://github.com/oven-sh/bun/issues/6686

**Comment by Micnubinub** [comment]
Those that need a temporary solution or a straight alternative you can try `surreal.node`, which is a binary that uses FFI

```js
const { Surreal } = require('surrealdb.node')
const surreal = new Surreal()
await surreal.connect('ws://127.0.0.1:8000/rpc', {
   auth: { user: 'user', pass: 'Password1!' },
})
await surreal.use({ ns: 'test', db: 'test' })
```

*Note that you may have to use `npm` to install instead of `bun` as I am still getting this error:*
<img width="1203" alt="image" src="https://github.com/oven-sh/bun/assets/4921356/988409cc-f916-4ba9-a103-bb7ed9fe9824">

**Comment by Electroid** [comment]
Duplicate of #6686

**Comment by happysalada** [comment]
@ikwr this should be fixed with 1.0.15 , if you have time to test, I'm curious.

**Comment by Micnubinub** [comment]
Yea, I'll be testing it as soon as 1.1 is released... Also curious which one will perform better `.js` or `.node`. Will post an update later

> @ikwr this should be fixed with 1.0.15 , if you have time to test, I'm curious.
